tests/test_cyto_utils/test_DeepProfiler_processing.py

```diff
@@ -19,92 +20,87 @@
 from pycytominer import normalize
 from pycytominer.cyto_utils import infer_cp_features
 
-tmpdir = tempfile.gettempdir()
-random.seed(42)
-# setting the file locations
-example_project_dir = os.path.join(
-    os.path.dirname(__file__), "..", "test_data", "DeepProfiler_example_data"
-)
-
-profile_dir = os.path.join(example_project_dir, "outputs", "results", "features")
-
-index_file = os.path.join(example_project_dir, "inputs", "metadata", "test_index.csv")
-
-output_folder = os.path.join(tmpdir, "DeepProfiler")
-os.makedirs(output_folder, exist_ok=True)
-
-
-deep_data = DeepProfilerData(
-    index_file=index_file,
-    profile_dir=profile_dir,
-)
-
-# compile single cell data from DP run
-single_cells_DP = SingleCellDeepProfiler(deep_data=deep_data)
-single_cells = single_cells_DP.get_single_cells(output=True)
-
-# normalize single cell data with DP processing
-output_file = os.path.join(output_folder, "normalized.csv")
-single_cells_normalized = single_cells_DP.normalize_deep_single_cells(
-    output_file=output_file
-)
-
-# normalize single cell data with CP processing
-# extract metadata prior to normalization
-metadata_cols = infer_cp_features(single_cells, metadata=True)
-# locations are not automatically inferred with cp features
-metadata_cols.append("Location_Center_X")
-metadata_cols.append("Location_Center_Y")
-derived_features = [x for x in single_cells.columns.tolist() if x not in metadata_cols]
-
-# wrapper for pycytominer.normalize() function
-CP_single_cells_normalized = normalize(
-    profiles=single_cells,
-    features=derived_features,
-)
-x_locations = single_cells["Location_Center_X"]
-CP_single_cells_normalized.insert(0, "Location_Center_X", x_locations)
-y_locations = single_cells["Location_Center_Y"]
-CP_single_cells_normalized.insert(1, "Location_Center_Y", y_locations)
-
-# calculating the dataframe for each depth
-site_class = AggregateDeepProfiler(
-    deep_data=deep_data,
-    aggregate_operation="median",
-    aggregate_on="site",
-)
-df_site = site_class.aggregate_deep()
-
-well_class = AggregateDeepProfiler(
-    deep_data=deep_data,
-    aggregate_operation="median",
-    aggregate_on="well",
-    output_file=output_folder,
-)
-df_well = well_class.aggregate_deep()
 
-plate_class = AggregateDeepProfiler(
-    deep_data=deep_data,
-    aggregate_operation="median",
-    aggregate_on="plate",
-    output_file=output_folder,
-)
-df_plate = plate_class.aggregate_deep()
+def test_DeepProfiler():
```

Comment:
Seeing the changes here and below made me wonder what might be lost from more modular testing given the move one larger test (I may be missing context here on design decisions surrounding these changes, and if so, apologies!). Consider retaining the modularity from the existing test functions instead of removing them and using a [Pytest fixture](https://docs.pytest.org/en/6.2.x/fixture.html) to help ensure the data used by these functions is provided from one common declarative spot.

Reply:
@d33bs I agree that using a pytest fixture is the correct approach here. I spent some time trying to implement that until I realized that it will require a complete restructuring of the entire set of tests in this file. I think we should add a note about the fixture approach in #304.  

The way an entire data package was generated outside of functions and functions only included assert statements about that data package means that these changes did not actually decrease real modular testing.

Reply:
Thank you for the details here; I better understand the context here. We do lose pytest test modularity here; there is now one large test performed with many asserts as opposed to smaller tests with fewer asserts. One drawback of this is the reduction of testing specificity during development (where we only want to test one failing test to conserve time, for example). This may make maintenance and test troubleshooting more difficult in the future.

I also wasn't sure and might have missed it: have we addressed or documented the root cause of the reason the tests were failing for `test_DeepProfiler_processing.py` to help justify the reasons for switching to the new approach found in this PR?

Generally, I'd still recommend a fixture approach here if possible to help retain the modularity and resolve the challenges mentioned in #304. It might be that [fixture scope](https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session) and multi-value / tuple returns from a fixture could help.

Reply:
I have changed the description of #304 to represent the primary bug which is being addressed here, which is the crashing of pytest during test collection. 

I created a separate [issue ](https://github.com/cytomining/pycytominer/issues/313) for the crashes that occur on smaller machines which will require more substantial re-factoring of `test_DeepProfiler_processing.py`.

Reply:
This re-factoring is completed done.

Reply:
Thank you @kenibrewer for the diligence on these items!